op-node/withdrawals/utils.go

```diff
@@ -163,9 +145,9 @@ type FinalizedWithdrawalParameters struct {
 // FinalizeWithdrawalParameters queries L2 to generate all withdrawal parameters and proof necessary to finalize an withdrawal on L1.
 // The header provided is very important. It should be a block (timestamp) for which there is a submitted output in the L2 Output Oracle
 // contract. If not, the withdrawal will fail as it the storage proof cannot be verified if there is no submitted state root.
-func FinalizeWithdrawalParameters(ctx context.Context, l2client ProofClient, txHash common.Hash, header *types.Header) (FinalizedWithdrawalParameters, error) {
+func FinalizeWithdrawalParameters(ctx context.Context, proofCl ProofClient, l2ReceiptCl ReceiptClient, txHash common.Hash, header *types.Header) (FinalizedWithdrawalParameters, error) {
```

Comment:
Half of this PR seems to be about changing the interface of this function.

Reply:
TBH I like this interface better - the old one required an RPC client to be instantiated, which was annoying because Geth doesn't expose the underlying RPC from the `ethclient.Client` and we always had to instantiate a fresh one :(

Reply:
fair, though making a new `gethclient` (the concrete type that satisifies `ProofClient`) still is usually done with an RPC client.

Reply:
The old coupled client made it impossible to instrument (required the concrete type instead of client.RPC interface) and didn't fit into testing for the same reason, hence the changes.

Reply:
So could a `client.RPC` have been used in the ProofClient struct?

Reply:
Not in the `Client` struct, since we need the proof and receipt methods. Instead this now uses the bare minimum interface, and separates the proof and receipt sources so the user can choose the implementation (for instrumentation, mocking, whatever that may be).